data/collector.go

```diff
@@ -93,6 +92,7 @@ func (c *collector) Close() {
 	c.logRoutine.Wait()
 	//nolint:errcheck
 	_ = c.logger.Sync()
+	c.closed = true
```

Comment:
Explaining more in the PR-level comment, but I would extra and extra `if c.closed { return }` check right after c.lock.Lock()

Reply:
Oh, good catch! Thanks for going through this! Will add ASAP